experimental/algorithm/LAGraph_RegularPathQuery.c

```diff
@@ -0,0 +1,357 @@
+    const GrB_Index *S,         // source vertices to start searching paths
+    size_t ns,                  // number of source vertices
+    char *msg                   // LAGraph output message
+)
+{
+
+    //--------------------------------------------------------------------------
+    // check inputs
+    //--------------------------------------------------------------------------
+
+    LG_CLEAR_MSG ;
+
+    GrB_Matrix frontier = NULL ;         // traversal frontier representing
+                                         // correspondence between NFA states
+                                         // and graph vertices
+    GrB_Matrix symbol_frontier = NULL ;  // part of the new frontier for the
+                                         // specific label
+    GrB_Matrix next_frontier = NULL ;    // frontier value on the next
+                                         // traversal step
+    GrB_Matrix visited = NULL ;          // visited pairs (state, vertex)
+    GrB_Vector final_reducer = NULL ;    // auxiliary vector for reducing the
+                                         // visited matrix to an answer
+
+    GrB_Index ng = 0 ;                   // # nodes in the graph
+    GrB_Index nr = 0 ;                   // # states odes in the NFA
+    GrB_Index states = ns ;              // # pairs in the current
+                                         // correspondence between the graph and
+                                         // the NFA
+
+    GrB_Index rows = 0 ;                 // utility matrix row count
+    GrB_Index cols = 0 ;                 // utility matrix column count
+    GrB_Index vals = 0 ;                 // utility matrix value count
+
+    GrB_Matrix *A = NULL ;
+    GrB_Matrix *B = NULL ;
+    GrB_Matrix *BT = NULL ;
+
+    LG_ASSERT (reachable != NULL, GrB_NULL_POINTER) ;
+    LG_ASSERT (G != NULL, GrB_NULL_POINTER) ;
+    LG_ASSERT (R != NULL, GrB_NULL_POINTER) ;
+    LG_ASSERT (S != NULL, GrB_NULL_POINTER) ;
+
+    (*reachable) = NULL ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (G[i] == NULL) continue ;
+        LG_TRY (LAGraph_CheckGraph (G[i], msg)) ;
+    }
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (R[i] == NULL) continue ;
+        LG_TRY (LAGraph_CheckGraph (R[i], msg)) ;
+    }
+
+    LG_TRY (LAGraph_Malloc ((void **) &A, nl, sizeof (GrB_Matrix), msg)) ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (G[i] == NULL) 
+        {
+            A[i] = NULL ;
+            continue ;
+        }
+
+        A[i] = G[i]->A;
+    }
+
+    LG_TRY (LAGraph_Malloc ((void **) &B, nl, sizeof (GrB_Matrix), msg)) ;
+    LG_TRY (LAGraph_Malloc ((void **) &BT, nl, sizeof (GrB_Matrix), msg)) ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (R[i] == NULL)
+        {
+            B[i] = NULL ;
+            continue ;
+        }
+
+        B[i] = R[i]->A;
+        if (R[i]->kind == LAGraph_ADJACENCY_UNDIRECTED ||
```

Comment:
I think, such a knowledge about the input should be described in comments (on arguments.)

Reply:
This code has actually been copy-pasted from the BFS implementation.

I've removed undirected check, since it's really unlikely for the NFA to be represented by an undirected graph. 
```
@@ -195,8 +195,7 @@ int LAGraph_RegularPathQuery
         }

         B[i] = R[i]->A ;
-        if (R[i]->kind == LAGraph_ADJACENCY_UNDIRECTED ||
-            R[i]->is_symmetric_structure == LAGraph_TRUE)
+        if (R[i]->is_symmetric_structure == LAGraph_TRUE)
         {
             BT[i] = B[i] ;
         }
```

Thank you.